draft-ietf-oauth-browser-based-apps.md

```diff
@@ -367,6 +367,37 @@ the necessary CORS headers to enable the JavaScript code to make this POST reque
 from the domain on which the script is executing. (See {{cors}} for additional details.)
 
 
+### Tokens Securely Handled by a Service worker
+In this scenario, a [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API) is responsible for obtaining tokens from the authorization server and passing them to the resource API.
+
+Service workers are inherently safe from XSS, because the browser APIs allowing to register one take an origin-constrained URL.
+They can thus be used as a safe store for tokens.
+
+Service workers can also be used to intercept calls from the frontend. As such, they can completely isolate calls to the authorization server from XSS attack surface.
+
+                                                             Resource               Authorization
+      User       Application    Service Worker                server                   server
+       |   browse     |               |                          |                        |
+       | ------------>|               |                          |          /authorize    |
+       |              |------------------------------------------------------------------->
```

Comment:
I think this line needs to be broken into two steps, to show that the application is asking the service worker to initiate the authorization request.